ci-operator/step-registry/workers-rhel/pre-hook/beta-repos/workers-rhel-pre-hook-beta-repos-commands.sh

```diff
@@ -0,0 +1,106 @@
+set -o pipefail
+set -x
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+export OPS_MIRROR_KEY=${CLUSTER_PROFILE_DIR}/ops-mirror.pem
+
+echo "PLATFORM_VERSION: '${PLATFORM_VERSION}'"
+
+# Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+# to be able to SSH.
+if ! whoami &> /dev/null; then
+    if [[ -w /etc/passwd ]]; then
+        echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    else
+        echo "/etc/passwd is not writeable, and user matching this uid is not found."
+        exit 1
+    fi
+fi
+
+oc version --client
+
+MIRROR_USERNAME="$(<'/var/run/mirror-repo-basic-auth/username')"
+export MIRROR_USERNAME
+MIRROR_PASSWORD="$(<'/var/run/mirror-repo-basic-auth/password')"
+export MIRROR_PASSWORD
+
+cat > prep-beta.yaml <<-'EOF'
+---
+- name: Prep Playbook
+  hosts: new_workers
+  any_errors_fatal: true
+  gather_facts: false
+
+  vars:
+    kubeconfig_path: "{{ lookup('env', 'KUBECONFIG') }}"
+    ops_mirror_path: "{{ lookup('env', 'OPS_MIRROR_KEY') }}"
+    platform_version: "{{ lookup('env', 'PLATFORM_VERSION') }}"
+    mirror_username: "{{ lookup('env', 'MIRROR_USERNAME') }}"
+    mirror_password: "{{ lookup('env', 'MIRROR_PASSWORD') }}"
+    major_platform_version: "{{ platform_version[:1] }}"
+
+  tasks:
+  - name: Get cluster version
+    command: >
+      oc get clusterversion
+      --kubeconfig={{ kubeconfig_path }}
+      --output=jsonpath='{.items[0].status.desired.version}'
+    delegate_to: localhost
+    register: oc_get
+    until:
+    - oc_get.stdout != ''
+
+  - name: Set release_version to cluster version
+    set_fact:
+      release_version: "{{ oc_get.stdout | regex_search('^\\d+\\.\\d+') }}"
+
+  - name: Wait for host connection to ensure SSH has started
+    wait_for_connection:
+      timeout: 600
+
+  - name: Create rhel-X-server-beta-rpms repo file
+    template:
+      src: "rhel-{{ major_platform_version }}-server-beta-rpms.repo.j2"
+      dest: "/etc/yum.repos.d/rhel-{{ major_platform_version }}-server-beta-rpms.repo"
+
+EOF
+
+# NOTE: Add rhel 7 beta repo if needed
+
+cat > rhel-8-server-beta-rpms.repo.j2 <<-'EOF'
+[rhel-8-for-x86_64-beta-baseos-rpms]
+name = Red Hat Enterprise Linux 8 for x86_64 Beta - BaseOS (RPMs)
+baseurl = https://mirror2.openshift.com/enterprise/reposync/ci-deps/rhel-8-beta-baseos-rpms/
+username = {{ mirror_username }}
+password = {{ mirror_password }}
+gpgcheck = 0
+sslverify = 0
+enabled = 1
+metadata_expire = 86400
+enabled_metadata = 1
+module_hotfixes = 1
+
+[rhel-8-for-x86_64-beta-appstream-rpms]
+name = Red Hat Enterprise Linux 8 for x86_64 Beta - AppStream (RPMs)
+baseurl = https://mirror2.openshift.com/enterprise/reposync/ci-deps/rhel-8-beta-appstream-rpms/
+username = {{ mirror_username }}
+password = {{ mirror_password }}
+gpgcheck = 0
+sslverify = 0
+enabled = 1
+metadata_expire = 86400
+enabled_metadata = 1
+module_hotfixes = 1
+EOF
+
+if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
```

Comment:
I guess we do not need this block.

Reply:
I thought this was for disconnected install behind proxy but not sure

Reply:
Actually most code comes from https://github.com/openshift/release/blob/master/ci-operator/step-registry/workers-rhel/repos/workers-rhel-repos-commands.sh

Reply:
Yeah, I see.

The proxy is used when running oc command against cluster api, especially for private cluster. In https://github.com/openshift/release/pull/38280/files#r1166696645, I proposed to remove "oc command" part,  because we are placing a rhel repo, not care about cluster version. Once that, we can drop all the lines prepared for running oc command lines.